### PR TITLE
Code freeze hook

### PR DESCRIPTION
**Pull request by asheepapart**
Ability to freeze a configured repository or set of repositories. 

@nadeemahmad and Robert Ansel, I'd be interested in your feedback.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @asheepapart has signed the CLA. Thanks for the pull request!

File: src/Bart/GitHook/CodeFreeze.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
+<?php
+/**
```
Comment (nadeemahmad) [inline src/Bart/GitHook/CodeFreeze.php]: Please remove this.

File: src/Bart/GitHook/CodeFreeze.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * Prevent changes when code is frozen
+ */
+class CodeFreeze extends GitHookAction
+{
+    /** @var GitHookSystemConfig */
+    private $config;
+
+    public function  __construct()
+    {
+        $this->config = Diesel::create('\Bart\GitHook\GitHookSystemConfig');
```
Comment (nadeemahmad) [inline src/Bart/GitHook/CodeFreeze.php]: This can potentially be part of the parent class.

File: src/Bart/GitHook/GitHookSystemConfig.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+;; This section holds names for environment variables to be used by hooks
+[env_vars]
+; How to determine name of user pushing a change
+push_user_name = GITOSIS_USER
+
+
+[frozen]
+; Do not all merges to the following repos
+; valid options are '', 'all' for everything, or a CSV of repo names
+repo_names = bart.git, example.git, my-organization/repo.git
```
Comment (nadeemahmad) [inline src/Bart/GitHook/GitHookSystemConfig.php]: What are the paths relative to? There should be a mention of that here.

File: src/Bart/GitHook/GitHookSystemConfig.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    public function README()
+    {
+        return <<<README
+;;
+; Configurations shared by all repositories on this host
+;;
+;; This section holds names for environment variables to be used by hooks
+[env_vars]
+; How to determine name of user pushing a change
+push_user_name = GITOSIS_USER
```
Comment (nadeemahmad) [inline src/Bart/GitHook/GitHookSystemConfig.php]: Can this default to `USER`?

File: src/Bart/GitHook/CodeFreeze.php (hunk 2 above)
Comment (asheepapart) [inline src/Bart/GitHook/CodeFreeze.php]: True. I'll wait until that happens before pulling it up. I prefer to not saturate signatures until necessary.

**Pull request merged**